app/api/microlocations.py

```diff
@@ -70,24 +81,15 @@ def query(self, view_kwargs):
             query_ = query_.join(Session).filter(Session.id == session.id)
         return query_
 
-    def before_create_object(self, data, view_kwargs):
-        event = None
-        if view_kwargs.get('event_id'):
-            event = safe_query(self, Event, 'id', view_kwargs['event_id'], 'event_id')
-        elif view_kwargs.get('event_identifier'):
-            event = safe_query(self, Event, 'identifier', view_kwargs['event_identifier'], 'event_identifier')
-        if event:
-            data['event_id'] = event.id
-
     view_kwargs = True
     decorators = (api.has_permission('is_coorganizer', fetch='event_id', fetch_as="event_id", methods="POST",
```

Comment:
I don't there is any requirement to set this whole concept. You can use permission manager directly for it. Since you removed POST method from this resource this means only direct endpoint will take care of adding a resource

To achieve the same what you can do is 
`decorators = (api.has_permission('is_coorganizer', fetch='event_id', fetch_as="event_id", methods="GET",....`
`methods = ['GET', 'POST', ]`
```
@jwt_required
def before_post(self, args, kwargs, data):
          require_relationship(['event'], data)
          if not has_access('is_coorganizer', event_id=data['event'])
              raise ForbiddenException(''.....")
```